### PR TITLE
Don't allow duplicate formats in configuration.

### LINKED ISSUE #683: Configuration.Default.ImageFormats shows duplicates
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
The collection returned by Configuration.Default.ImageFormats shows duplicate ImageFormta instances.

### Steps to Reproduce
```c#
        private static void ShowConfigurationDefaultImageFormats()
        {
            Console.WriteLine($"Executing {System.Reflection.MethodBase.GetCurrentMethod().Name}");

            Console.WriteLine("Image Formats: ");

            foreach(IImageFormat F in Configuration.Default.ImageFormats)
            {
                Console.Write($"  {F}  Mime Type: {F.DefaultMimeType}  Extension(s):  ");
                foreach(string ext in F.FileExtensions)
                {
                    Console.Write($"{ext}  ");
                }
                Console.WriteLine();
            }

        }
```
Produces:
```
Image Formats:
  SixLabors.ImageSharp.Formats.Bmp.BmpFormat  Mime Type: image/bmp  Extension(s):  bm  bmp  dip
  SixLabors.ImageSharp.Formats.Bmp.BmpFormat  Mime Type: image/bmp  Extension(s):  bm  bmp  dip
  SixLabors.ImageSharp.Formats.Gif.GifFormat  Mime Type: image/gif  Extension(s):  gif
  SixLabors.ImageSharp.Formats.Gif.GifFormat  Mime Type: image/gif  Extension(s):  gif
  SixLabors.ImageSharp.Formats.Jpeg.JpegFormat  Mime Type: image/jpeg  Extension(s):  jpg  jpeg  jfif
  SixLabors.ImageSharp.Formats.Jpeg.JpegFormat  Mime Type: image/jpeg  Extension(s):  jpg  jpeg  jfif
  SixLabors.ImageSharp.Formats.Png.PngFormat  Mime Type: image/png  Extension(s):  png
  SixLabors.ImageSharp.Formats.Png.PngFormat  Mime Type: image/png  Extension(s):  png
```

### System Configuration

- ImageSharp version: 1.0.0-beta0005
- Other ImageSharp packages and versions: SixLabors.ImageSharp.Drawing; SixLabors.ImageSharp
- Environment (Operating system, version and so on): Win 10; Visual Studio 2017 Enterprise 15.8.0
- .NET Framework version: .Net Core 2.1
- Additional information:

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Fixes #683 

`ConcurrentBag<T>` allows duplicates which we do not want. I locked a `HashSet<T>` since there is no `ConcurrentSet<T>` type in the .NET Framework.

### CODE REVIEW

**Comment by tzachshabtay** [comment]
In my project I used ConcurrentDictionary with a byte value which I ignore: https://github.com/tzachshabtay/MonoAGS/blob/master/Source/Engine/AGS.Engine/Misc/Collection/AGSConcurrentHashSet.cs

I don't know if it's better than simply locking the hashset.

File: tests/ImageSharp.Tests/ConfigurationTests.cs [hunk 1]
```diff
@@ -96,5 +92,13 @@ public void AddFormatCallsConfig()
 
             provider.Verify(x => x.Configure(config));
         }
+
+        [Fact]
+        public void DefaultConfigurationHasCorrectFormatCount()
```
[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply tests/ImageSharp.Tests/ConfigurationTests.cs]: Yeah I'll add a simple one.

**Comment by JimBobSquarePants** [comment]
@tzachshabtay I think I might do that actually; the lock felt ugly. Thanks!

**Comment by antonfirsov** [comment]
Why do we even try adding duplicates? Maybe we should just remove that part of code.

**Comment by JimBobSquarePants** [comment]
We call it when adding both encoders and decoders since they are independent operations

**Pull request merged**